components/raftstore/src/store/fsm/apply.rs

```diff
@@ -3073,6 +3097,11 @@ pub struct Apply<C> {
     pub entries_size: usize,
     pub cbs: Vec<Proposal<C>>,
     pub bucket_meta: Option<Arc<BucketMeta>>,
+    // Used for witness, as witness would skip applying write commands, so the applied index in the
```

Comment:
Why not always update the applied index?

Reply:
the write commands are not sent to apply worker if it's a witness so the apply system can save CPU time to poll that fsm.